can/exceptions.py

```diff
@@ -0,0 +1,61 @@
+"""
+There are several specific :class:`Exception` classes to allow user
+code to react to specific scenarios related to CAN busses::
+
+    Exception (Python standard library)
+     +-- ...
+     +-- CanError (python-can)
+         +-- CanBackendError
+         +-- CanInitializationError
+         +-- CanOperationError
+         +-- CanTimeoutError
+
+Keep in mind that some functions and methods may raise different exceptions.
+For example, validating typical arguments and parameters might result in a
+:class:`ValueError`. This should be documented for the function at hand.
+"""
+

```

Comment:
TODO for me: [Add return codes where applicable](https://github.com/hardbyte/python-can/pull/562#discussion_r283294876).